### PR TITLE
limiter: don't return a negative amount of allowed bytes

### PR DESCRIPTION
**Pull request by nickvanw**
The bug looks like this:

1. We have a pool where we're allowed to write 10MB in 1 second
2. We have two writers, 1 and 2. Each of them can write 5MB this period. 
3. Each of them write 4 MB.
4. We borrow another writer, 3. 
5. We assign, in `limiter.go` the amount that writers 1 and 2 are allowed to write in this period. It's not 5MB any more, it's 3.33MB. 
6. Writer 1 calls for the amount it's allowed to write, which is now negative .66 MB. This causes a panic.

Basically, this results from the amount of data we _actually_ sent being more  than what we thought we were allowed to do in this period, so we would return a negative amount for what we are allowed to do next time. 

@macb

### CODE REVIEW

**Comment by aybabtme** [comment]
if this fix is in prod, a regression test would be cool @nickvanw

**Comment by macb** [comment]
ya we hot patched the vendor'd version.

**Comment by nickvanw** [comment]
Added a small regression test, without this PR the result is
```
--- FAIL: TestLimiterCanDo (0.00s)
	limiter_test.go:14: wanted to be able to write nothing, got: -1048576
```

**Comment by aybabtme** [comment]
:rocket:
